Course/Course/Presentation/Subviews/CourseVideoDownloadBarView/CourseVideoDownloadBarView.swift

```diff
@@ -135,13 +135,15 @@ struct CourseVideoDownloadBarView: View {
         Toggle("", isOn: .constant(viewModel.isOn))
```

Comment:
@rnr did you try to refactor that code and use @Published property here...
so in view model:
```swift
@Published var isOn:Bool = false{
  didSet {
      if !isInternetAvaliable {
         onNotInternetAvaliable?()
         return
     }
     Task { await onToggle()  }
  }
}
```

and in view:
```swift
Toggle("", isOn: $viewModel.isOn)
  .toggleStyle(SwitchToggleStyle(tint: Theme.Colors.toggleSwitchColor))
  .padding(.trailing, 15)
  .accessibilityIdentifier("download_toggle")
```

Is there any reason why we are using TapGesture here?

Reply:
@forgotvas  The reason is that Toggle is not used here as a classic Toggle. The user interacts with the Toggle and we show an Alert where we have to decide whether the Toggle should change its state or not.

Reply:
got it